tests/integration/goal_oriented/test_write_file.py

```diff
@@ -28,19 +25,12 @@
 
 
 @requires_api_key("OPENAI_API_KEY")
-def test_write_file() -> None:
```

Comment:
The introduction of a pytest fixture that does its own cleanup removes a ton of boilerplate setup/teardown from tests